docs/design.md

```diff
@@ -0,0 +1,104 @@
+# Design and architecture
+
+## Design
+The `image-rs` crate is a rustified and tailored version of [containers/image](https://github.com/containers/image),
+to provide a small, simple, secure, lightweight and high performance OCI container image management library for
+[Confidential Containers](https://github.com/confidential-containers).
+
+Currently the `image-rs` crate focuses on usage scenarios to run `Confidential Containers` on Kubernetes clusters with
+[Kata Containers](https://katacontainers.io/). It may also be enhanced to support other confidential container runtimes
+on demand. So it implements the
+[CRI Image Service](https://github.com/kubernetes/cri-api/blob/master/pkg/apis/runtime/v1alpha2/api.proto#L119)
+interface to be easily integrated with the K8s ecosystem. Among the container image `Build`, `Ship` and `Run` stages,
+it focuses on the `Run` stage instead of `Build` and `Ship`, that is only to provide interfaces to prepare container
+images for running in a confidential environment. Users may use any OCI image/distribution spec compatible tools to
+build and distribute encrypted/plaintext container images with or without signatures.
+
+The `image-rs` crate supports both encrypted and unencrypted container images since
```

Comment:
I'd consider switching the order of these image types as it's the encrypted type that is "mandatory for many confidential container usage scenarios":

```suggestion
The `image-rs` crate supports both unencrypted and encrypted container images since
```

Reply:
will switch the order.